src/amy-piano.c

```diff
@@ -1,8 +1,7 @@
 // amy-example.c
 // a simple C example that plays audio using AMY out your speaker 
+#if !(defined(ESP_PLATFORM) || defined(PICO_ON_DEVICE) || defined(ARDUINO) || defined(__IMXRT1062__) || defined(ARDUINO_ARCH_RP2040) ||defined(ARDUINO_ARCH_RP2350))
 
-// This doesn't compile under ARDUINO because of no usleep.  Arduino tries to compile everything anyway.
-#ifndef ARDUINO
```

Comment:
I think the only problem is ARDUINO.  It's true we don't want to make this file on the other MCU platforms, but we can achieve that through makefile rules (i.e., not including it).  It's only a problem for Arduino where there is no Makefile and the environment insists on compiling *every* *.c file in the src/ directory.